aptos-move/framework/aptos-framework/sources/transaction_validation.move

```diff
@@ -185,30 +271,33 @@ module aptos_framework::transaction_validation {
                 }
             };
         };
+    }
 
-        let max_transaction_fee = txn_gas_price * txn_max_gas_units;
+    // Question[Orderless]: Is this logic correct?
+    fun check_for_replay_protection_orderless_txn(
+        sender: address,
+        nonce: u64,
+        txn_authentication_key: Option<vector<u8>>,
+        txn_expiration_time: u64,
+        is_simulation: bool,
+    ) {
+        assert!(
+            txn_expiration_time <= timestamp::now_seconds() + MAX_EXPIRATION_TIME_FOR_ORDERLESS_TXNS,
+            error::invalid_argument(PROLOGUE_ETRANSACTION_EXPIRATION_TOO_FAR_IN_FUTURE),
+        );
+        assert!(nonce_validation::check_and_insert_nonce(sender, nonce, txn_expiration_time), error::invalid_argument(PROLOGUE_ENONCE_ALREADY_USED));
 
-        if (!features::transaction_simulation_enhancement_enabled() || !skip_gas_payment(
-            is_simulation,
-            gas_payer_address
-        )) {
-            assert!(
-                permissioned_signer::check_permission_capacity_above(
-                    gas_payer,
-                    (max_transaction_fee as u256),
-                    GasPermission {}
-                ),
-                error::permission_denied(PROLOGUE_PERMISSIONED_GAS_LIMIT_INSUFFICIENT)
-            );
-            if (features::operations_default_to_fa_apt_store_enabled()) {
+        if (!features::transaction_simulation_enhancement_enabled() ||
+                !skip_auth_key_check(is_simulation, &txn_authentication_key)) {
+            if (option::is_some(&txn_authentication_key)) {
                 assert!(
-                    aptos_account::is_fungible_balance_at_least(gas_payer_address, max_transaction_fee),
-                    error::invalid_argument(PROLOGUE_ECANT_PAY_GAS_DEPOSIT)
+                    txn_authentication_key == option::some(account::get_authentication_key(sender)),
```

Comment:
why this part is different for orderless txn?

Reply:
They are same. I moved the assertion to a common place.